cmd/ipfs/main.go

```diff
@@ -604,63 +603,7 @@ func getApiClient(repoPath, apiAddrStr string) (cmdsHttp.Client, error) {
 		return nil, err
 	}
 
-	client, err := apiClientForAddr(addr)
-	if err != nil {
-		return nil, err
-	}
-
-	// make sure the api is actually running.
-	// this is slow, as it might mean an RTT to a remote server.
-	// TODO: optimize some way
```

Comment:
ok, there's a problem here. part of the goal is to _check_ that the endpoint is available, which may be a case of a stale `$repo/api` file (eg ipfs daemon, then kill -9 $pid). we could "check" by trying to use it, but im no longer sure how the flow will work here. I recall it being tricky to get right + cover all the bases.